app/knights/knight.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from app.knights.inventory.weapon import Weapon
+from app.knights.inventory.armour import Armour
+from app.knights.inventory.consumables import Potion
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:
```

Comment:
Here as well would be better to have many params instead of `knight: dict` - this way you make it obvious which params are mandatory and which ones are optional.